TowerForge/TowerForge/GameModule/PowerUps/InvulnerabilityPowerUp.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  InvulnerabilityPowerUp.swift
+//  TowerForge
+//
+//  Created by Keith Gan on 31/3/24.
+//
+
+class InvulnerabilityPowerUp: EventTransformation {
+    func transformEvent(event: TFEvent) -> TFEvent {
+        guard var damageEvent = event as? DamageEvent, damageEvent.player == Player.oppositePlayer else {
+            return event
+        }
+
+        damageEvent.toggleActive()
```

Comment:
I think it will be better to just return nil here

Reply:
Removed.